packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_item_details/blocs/item_details_bloc.dart

```diff
@@ -0,0 +1,68 @@
+{{> licence.dart }}
+
+import 'package:rx_bloc/rx_bloc.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../../base/common_services/item_service.dart';
+import '../../base/extensions/error_model_extensions.dart';
+import '../../base/models/errors/error_model.dart';
+import '../../base/models/item_model.dart';
+
+part 'item_details_bloc.rxb.g.dart';
+
+/// A contract class containing all events of the ItemDetailsBloC.
+abstract class ItemDetailsBlocEvents {
+  void getItemDetails(String itemId);
+}
+
+/// A contract class containing all states of the ItemDetailsBloC.
+abstract class ItemDetailsBlocStates {
+  /// The loading state
+  Stream<bool> get isLoading;
+
+  /// The error state
+  Stream<ErrorModel> get errors;
+
+  Stream<Result<ItemModel>> get item;
+}
+
+@RxBloc()
+class ItemDetailsBloc extends $ItemDetailsBloc {
+  ItemDetailsBloc(
+      this._itemService, {
+        required String itemId,
+        ItemModel? item,
+      }) : _itemSubject = BehaviorSubject<Result<ItemModel>>.seeded(
+      item != null ? Result.success(item) : Result.loading()) {
+    initItemData(itemId);
+  }
+
+  final ItemService _itemService;
+
+  final BehaviorSubject<Result<ItemModel>> _itemSubject;
```

Comment:
Do we mutade the _itemSubject ? If not please avoid creating Subjects within the class

Reply:
Reworked